src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -391,7 +391,15 @@ public JsonPropertyInfo GetProperty(ReadOnlySpan<byte> propertyName, ref ReadSta
             return new Dictionary<string, JsonPropertyInfo>(capacity, comparer);
         }
 
-        public JsonPropertyInfo? PolicyProperty { get; private set; }
+        /// <summary>
```

Comment:
Do we want to keep/state the reason why we need this? Previously it talked about `avoiding boxing`.

Reply:
Currently it says "is used to obtain the converter for the given type". This doesn't directly avoid boxing -- the strongly-typed converter model avoids boxing, so I didn't move that comment over. I can do another pass on these comments however.

FWIW an alternative to "policy property" is to add a `JsonConverter` property to `JsonClassInfo` but then we'd need to run a bunch of `if` statements like:
```cs
JsonConverter GetConverter()
{
    JsonConverter? converter = state.Current.JsonPropertyInfo?.Converter;
    if (converter == null)
    {
        converter = state.Current.JsonClassInfo?.Converter;
    }
    return converter!;
}
```

plus refactor the code to ensure every method that uses this has access to the `state` object here or both `JsonPropertyInfo` and `JsonClassInfo`.